src/Microsoft.DotNet.Interactive/Commands/Quit.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System;
+using System.Reactive.Disposables;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Interactive.Commands
+{
+    public class Quit : KernelCommand
+    {
+        private static readonly CompositeDisposable DisposeOnQuit = new();
+
+        public static void RegisterForDisposalOnQuit(IDisposable disposable)
```

Comment:
Why have potentially different disposal semantics than `Kernel.Dispose`? This looks like something that doesn't belong in the library. It should probably just be a host concern.

Reply:
we can have the host to register the OnQuit handler, and throw if it is not set